eng/common/TestResources/New-TestResources.ps1

```diff
@@ -377,7 +377,9 @@ try {
         $BaseName = 't' + (New-Guid).ToString('n').Substring(0, 16)
         Log "Generated base name '$BaseName' for CI build"
     } elseif (!$BaseName) {
-        $BaseName = "$UserName$ServiceDirectory"
+        # Handle service directories in nested directories, e.g. `data/aztables`
+        $serviceDirectorySafeName = $ServiceDirectory -replace '[/\\]', ''
```

Comment:
What cases do we hit this? I assume this is only for local runs without passing in a base directory but in those cases do we have a servicedirectory?

Reply:
I'm not sure what you mean? In the local run scenario you do have a service directory, e.g. for go it would be `data/aztables` which is relative to `sdk/data/aztables`. In CI it doesn't construct a semantically meaningful base name, whereas locally it tries to.